solr/solrj/src/java/org/apache/solr/common/cloud/ZkMaintenanceUtils.java

```diff
@@ -338,14 +338,14 @@ public static void downloadFromZK(SolrZkClient zkClient, String zkPath, Path fil
       if (children.size() == 0) {
         // If we didn't copy data down, then we also didn't create the file. But we still need a marker on the local
         // disk so create an empty file.
-        if (copyDataDown(zkClient, zkPath, file.toFile()) == 0) {
+        if (copyDataDown(zkClient, zkPath, file) == 0) {
           Files.createFile(file);
         }
       } else {
         Files.createDirectories(file); // Make parent dir.
         // ZK nodes, whether leaf or not can have data. If it's a non-leaf node and
         // has associated data write it into the special file.
-        copyDataDown(zkClient, zkPath, new File(file.toFile(), ZKNODE_DATA_FILE));
```

Comment:
Every time I saw this in the source I felt sad ;-)

Reply:
Yeah; me too!